modules/cudf/src/data_frame.ts

```diff
@@ -884,4 +884,39 @@ export class DataFrame<T extends TypeMap = any> {
     });
     return new DataFrame(series_map);
   }
+
+  /**
+   * Creates a DataFrame of `BOOL8` Series where `true` indicates the value is `null` or `NaN` and
+   * `false` indicates the value is valid.
+   *
+   * @returns DataFrame<Bool8> with series of `BOOL8` elements with `true` representing `null` or
+   * `NaN` values
+   *
+   * @example
+   * ```typescript
+   * import {DataFrame, Series, Int32, Float32}  from '@rapidsai/cudf';
+   * const df = new DataFrame({
+   *  a: Series.new({type: new Int32, data: [0, null, 2, null, 4]}),
+   *  b: Series.new({type: new Float32, data: [0, 1, 2, NaN, 4, NaN]})
+   * });
+   *
+   * df.isNull()
+   * // return {
+   * //    a: [false, true, false, true, false],
+   * //    b: [false, false, false, true, false, true],
+   * // }
+   * ```
+   */
+  isNull(): DataFrame<T> {
+    const temp       = new Table({columns: this.select(this.names)._accessor.columns});
+    const series_map = {} as SeriesMap<T>;
+    this._accessor.names.forEach((name, index) => {
+      if (this.get(name) instanceof Float32Series || this.get(name) instanceof Float64Series) {
+        series_map[name] = Series.new(temp.getColumnByIndex(index).isNaN()) as any;
+      } else {
+        series_map[name] = Series.new(temp.getColumnByIndex(index).isNull()) as any;
+      }
+    });
+    return new DataFrame(series_map);
+  }
```

Comment:
Here's how I'd implement this:

```suggestion
  public isNull() {
    return new DataFrame(new ColumnAccessor(this.names.reduce((cols, name) => {
      cols[name] = this.get(name).isNull()._col;
      return cols;
    }, {} as ColumnsMap<{[P in keyof T]: Bool8}>)));
  }
```

Reply:
yeah saw the `reduce` and `forEach` used in other methods in dataframe, assumed that floats needed to be handled differently so left it as is.